File: informatika_ismeretek/emelt/2017_may/java/IdozitettFelirat.java
import java.time.*;
import java.time.format.*;

public class IdozitettFelirat {
    public final String idozites;
    public final String felirat;

    public IdozitettFelirat(String idoz, String felir){
        idozites = idoz;
        felirat = felir;
    }

    public int szavakSzama(){
        return felirat.chars().filter(k -> k == ' ').map(k -> 1).sum() + 1;
    }

    private static String str_idoforma(String ido){
        return LocalTime.parse(ido, DateTimeFormatter.ISO_LOCAL_TIME.withResolverStyle(ResolverStyle.LENIENT))
                        .format(DateTimeFormatter.ofPattern("HH:mm:ss"));
    }

    public String strIdozites(){
        var split = idozites.split(" - ");
        return str_idoforma("00:" + split[0]) + "--> " + str_idoforma("00:" + split[1]);
    }
}

File: informatika_ismeretek/emelt/2017_may/c#/IdozitettFelirat.cs
using System.Linq;

public class IdozitettFelirat {

    public readonly string idozites;
    public readonly string felirat;

    public IdozitettFelirat(string idoz, string felir){
        idozites = idoz;
        felirat = felir;
    }

    public int SzavakSzama() => felirat.Where(k => k == ' ').Count() + 1;

    private string Str_idoforma(string ido) {
        var perc_mp = ido.Split(':');
        var pars = int.Parse(perc_mp[0]);

        return string.Format("{0:00}:{1:00}", pars / 60, pars % 60) + ":" + perc_mp[1];
    }

    public string StrIdozites() => Str_idoforma(idozites.Split('-')[0]) + "--> " + Str_idoforma(idozites.Split('-')[1]);
}